### ISSUE TITLE
[Windows] NetStream::dispose() when Video object added after NetStream::decodedFrames > 0 cause crash

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#860]
## Problem Description
`NetStream::dispose()` when `Video` object added after `NetStream::decodedFrames > 0` cause crash with Windows.
This is very tricky bug but it affect almost every AIR application that preload video, waiting for "first frame" and only afer that add it to stage. Such approach can avoid "flickering", switching video through "black screen" etc "negative" visual effects for users. Now it broken.
Crash happens more often if application not activated (in foucs) at `NetStream::dispose()` moment.

Tested with latests AIR 33.1.1.444 and AIR 33.1.1.476 with multiple Windows devices with different OS versions with 32-bit and 64-bit AIR applications with captive runtime with different videos.
Same problem in all cases.
There is no such crashes for AIR 32.0.0.89 and earlier.
Also there is no such crashes for other platforms.

There is no such problems with `StageVideo`.

## Steps to Reproduce
_Note: it one of the multiple ways to reproduce it but with that creepy code it reproduces in 100% times. In real application it almost unreal to be found source of this crash._
1) Launch code below with Windows device. It will playback video in loop.
2) In Windows tray (bottom right corner) you will see red square (sample application icon). Click on that and select the one element in context menu. Context menu here just to be sure that AIR application not active (not in focus) right now - in such situation crash can be easily reproduced (I don't know why).
Selecting context menu item dispose current `NetStream` (with all other resources) and start video playback again.
Sometimes you need to repeat this step 2-4 times to reproduce crash.

Application example with sources and example of video attached.
[air33_video_dispose_crash.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/6496449/air33_video_dispose_crash.zip)

```
package  {
	import flash.display.Sprite;
	import flash.events.Event;
	import flash.net.NetConnection;
	import flash.net.NetStream;
	import flash.media.Video;
	import flash.events.NetStatusEvent;
	import flash.events.MouseEvent;
	import flash.desktop.NativeApplication;
	import flash.desktop.SystemTrayIcon;
	import flash.events.ScreenMouseEvent;
	import flash.display.BitmapData;
	import flash.display.NativeMenu;
	import flash.display.NativeMenuItem;
	
	public class AIR33VideoDisposeCrash extends Sprite {
		private var video:Video = new Video(640, 480);
		private var nc:NetConnection;
		private var ns:NetStream;
		
		public function AIR33VideoDisposeCrash() {
			addEventListener(Event.ADDED_TO_STAGE, init);
		}
		
		private function init(e:Event):void {
			removeEventListener(Event.ADDED_TO_STAGE, init);
			
			if (NativeApplication.supportsSystemTrayIcon){//For Windows
				var trayIcon:SystemTrayIcon = NativeApplication.nativeApplication.icon as SystemTrayIcon;
				trayIcon.addEventListener(ScreenMouseEvent.CLICK, trayIcon_click);
				trayIcon.addEventListener(ScreenMouseEvent.RIGHT_CLICK, trayIcon_click);
				
				var _trayBasicIcons:Array = [new BitmapData(48, 48, false, 0xff0000)];
				(NativeApplication.nativeApplication.icon as SystemTrayIcon).bitmaps = _trayBasicIcons;
			}
			
			startPlayback();
		}
		
		private function startPlayback():void {
			nc = new NetConnection();
			nc.addEventListener(NetStatusEvent.NET_STATUS, ncHandler);
			nc.connect(null);
		}
		
		private function select(e:Event):void {
			removeEventListener(Event.ENTER_FRAME, enterFrame);
			removeChildren();
			
			video.attachNetStream(null);
			ns.removeEventListener(NetStatusEvent.NET_STATUS, nsHandler);
			ns.dispose();//This line cause crash!
			ns = null;
			
			nc.close();
			nc.removeEventListener(NetStatusEvent.NET_STATUS, ncHandler);
			nc = null;
			
			startPlayback();//Playback again
		}
		
		private function trayIcon_click(e:ScreenMouseEvent):void {
			var currentContextMenu:NativeMenu = new NativeMenu();
			currentContextMenu.items = [new NativeMenuItem("click")];
			currentContextMenu.addEventListener(Event.SELECT, select);
			(NativeApplication.nativeApplication.icon as SystemTrayIcon).menu = currentContextMenu;
			currentContextMenu.display(stage, e.screenX, e.screenY);
		}
		
		private function ncHandler(e:NetStatusEvent):void {
			if (e.info.code == "NetConnection.Connect.Success"){
				ns = new NetStream(nc);
				ns.client = {onMetaData:getMeta, onComplete:onComplete };
				ns.addEventListener(NetStatusEvent.NET_STATUS, nsHandler);
				
				addEventListener(Event.ENTER_FRAME, enterFrame);
				video.attachNetStream(ns);
				ns.play("neon.mp4");
			}
		}
		
		private function enterFrame(e:Event):void {
			if (ns.decodedFrames > 0){
				removeEventListener(Event.ENTER_FRAME, enterFrame);
				addChild(video);//Attach video to stage only at this moment
			}
		}
		
		private function nsHandler(e:NetStatusEvent):void {
			trace(e.info.code);
			if (e.info.code == "NetStream.Play.Stop"){
				removeEventListener(Event.ENTER_FRAME, enterFrame);
				video.attachNetStream(null);
				removeChildren();
				
				ns.removeEventListener(NetStatusEvent.NET_STATUS, nsHandler);
				ns.dispose();//Sometimes this line cause crash too
				ns = null;
				
				nc.close();
				nc.removeEventListener(NetStatusEvent.NET_STATUS, ncHandler);
				nc = null;
				
				startPlayback();//Playback again
			}
		}
		
		private function getMeta(mdata:Object):void { trace("meta"); }
		private function onComplete(mdata:Object):void { trace("onComplete"); }
		
	}
}
```

**Actual Result:**
Application crashes with error:
```
Application crashes with: air33_video_dispose_crash.exe, version: 0.0.0.0, time stamp: 0x608ab7da
Faulting module name: ntdll.dll, version: 10.0.19041.662, time stamp: 0x5f641e44
Exception code: 0xc0000374
Fault offset: 0x000e6ac3
Faulting process ID: 0x3cc
Faulting application start time: 0x01d74b47b1bb633b
Faulting application path: C:\air33_video_dispose_crash\air33_video_dispose_crash.exe
Faulting module path: C:\Windows\SYSTEM32\ntdll.dll
Report ID: 08041993-fc3f-4b01-a82e-1e113ecfc545
Faulting package full name: 
Faulting package-relative application ID: 

Problem signature:
P1: air33_video_dispose_crash.exe
P2: 0.0.0.0
P3: 608ab7da
P4: StackHash_94cf
P5: 10.0.19041.662
P6: 5f641e44
P7: c0000374
P8: PCH_26_FROM_ntdll+0x00071BDC
P9: 
P10: 
```

**Expected Result:**
Current video (with `NetStream` and other resources) will be disposed and start playback again without crashes.

## Known Workarounds
1) Use AIR 32.0.0.89.
2) Add `Video` object to stage before `NetStream::play()`, but it not always possible, especially for already used libraries and code.

**Comment by ajwfrost** [comment]
Thanks for the detailed info and reproduction steps.. This was a double-delete caused by issues with the decoder's inheritance/overriding of functions. Fix will be in our next release..

**Comment by itlancer** [comment]
It fixed with AIR 33.1.1.533 BETA.
Thanks! Waiting for AIR release version to close this issue.

**Comment by ajwfrost** [comment]
Should be fixed in 33.1.1.533, please check and re-open if necessary.
Thanks
